src/docker-osx-dev

```diff
@@ -440,13 +444,19 @@ function init_docker_host {
 
 #
 # Installs rsync on the Boot2Docker VM, unless it's already installed.
+# If the main repository is down, rsync will be installed from a mirror
+# If the mirror is down, download rsync binary and place in Boot2Docker VM
 #
 function install_rsync_on_docker_host {
   log_info "Installing rsync in the Docker Host image"
 
-  $DOCKER_HOST_SSH_COMMAND "if ! type rsync > /dev/null 2>&1; then tce-load -wi rsync; fi"
-}
+  $DOCKER_HOST_SSH_COMMAND "if ! type rsync > /dev/null 2>&1; then tce-load -wi rsync; fi" ||
+  $DOCKER_HOST_SSH_COMMAND "echo 'http://distro.ibiblio.org/' > /opt/tcemirror; tce-load -wi rsync;" ||
+  $DOCKER_HOST_SSH_COMMAND "sudo wget -P $BIN_DIR $RSYNC_BINARY_URL && sudo chmod +x $BIN_DIR/rsync"
 
+  # Reset tcemirror back to default
+  $DOCKER_HOST_SSH_COMMAND "echo 'http://repo.tinycorelinux.net/' > /opt/tcemirror"
```

Comment:
I think a safer approach is to backup the `/opt/tcemirror` file (e.g. copy it to `/opt/tcemirror.bak`) and then restore it. I'd also only do the backup/restore if the original mirror fails, and I'd log something to the console to indicate the backup and restore are happening to make debugging easier.